docs/bir-spec/src/test/resources/test-src/values.bal

```diff
@@ -175,26 +175,38 @@ function testNonNilNonMappingJsonMerge() returns boolean {
     json j2 = ["hello", "world"];
 
     json|error mj = j1.mergeJson(j2);
-    return mj is error && mj.message() == MERGE_JSON_ERROR_REASON &&
-        mj.detail()[MESSAGE].toString() == "Cannot merge JSON values of types 'float' and 'json[]'";
+    if (mj is error) {
+        var mjDetail = mj.detail()[MESSAGE];
+        string detailString = mjDetail is error ? mjDetail.toString() : mjDetail.toString();
+        return mj.message() == MERGE_JSON_ERROR_REASON &&
+            detailString == "Cannot merge JSON values of types 'float' and 'json[]'";
+    }
+    return false;
 }
 
 function testMappingJsonAndNonMappingJsonMerge1() returns boolean {
     json j1 = { one: "hello", two: "world", three: 1 };
     json j2 = "string value";
 
     json|error mj = j1.mergeJson(j2);
+    var mjDetail = mj is error ? mj.detail()[MESSAGE] : ();
+    string detailString = mjDetail is error ? mjDetail.toString() : mjDetail.toString();
     return mj is error && mj.message() == MERGE_JSON_ERROR_REASON &&
-        mj.detail()[MESSAGE].toString() == "Cannot merge JSON values of types 'map<json>' and 'string'";
+        detailString == "Cannot merge JSON values of types 'map<json>' and 'string'";
```

Comment:
In these scenarios `mj.detail()[MESSAGE]` is most probably a `string` value.

So we can probably just do

```ballerina
return mj is error && mj.message() == MERGE_JSON_ERROR_REASON &&
        <string> checkpanic mj.detail()[MESSAGE] == "Cannot merge JSON values of types 'map<json>' and 'string'";
```